### PR TITLE
pkp/pkp-lib#2894 Implement reviewer selection list panel

### PR DESCRIPTION
**Pull request by NateWr**

### CODE REVIEW

File: controllers/list/users/SelectReviewerListHandler.inc.php [hunk 1]
```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * @file controllers/list/users/SelectReviewerListHandler.inc.php
+ *
+ * Copyright (c) 2014-2018 Simon Fraser University
+ * Copyright (c) 2000-2018 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class SelectReviewerListHandler
+ * @ingroup classes_controllers_list
```
Comment (bozana) [inline controllers/list/users/SelectReviewerListHandler.inc.php]: correct?

File: controllers/list/users/SelectReviewerListHandler.inc.php [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+			'daysSinceLastAssignmentDescription' => __('reviewer.list.daysSinceLastAssignmentDescription'),
+			'averageCompletion' => __('reviewer.list.averageCompletion'),
+			'neverAssigned' => __('reviewer.list.neverAssigned'),
+			'currentlyAssigned' => __('reviewer.list.currentlyAssigned'),
+			'reviewInterests' => __('reviewer.list.reviewInterests'),
+			'gossip' => __('user.gossip'),
+			'biography' => __('reviewer.list.biography'),
+			'listSeparator' => __('common.listSeparator'),
+			'viewMore' => __('review.list.viewMore'),
+			'viewLess' => __('review.list.viewLess'),
```
Comment (bozana) [inline controllers/list/users/SelectReviewerListHandler.inc.php]: Those should be called "reviewer.list...." 
Could they maybe be put together with those above submission.list.viewMore/less i.e. have something like common.list.viewMore/Less?
Some of the locale keys here also contain variables, that are not provided here, or?

File: controllers/list/submissions/PKPSubmissionsListHandler.inc.php [hunk 3]
```diff
@@ -139,11 +121,12 @@ public function getConfig() {
...
 			'no' => __('common.no'),
 			'deleting' => __('common.deleting'),
+			'currentStage' => __('submission.list.currentStage'),
 			'confirmDelete' => __('submission.list.confirmDelete'),
 			'responseDue' => __('submission.list.responseDue'),
 			'reviewDue' => __('submission.list.reviewDue'),
-			'filter' => __('submission.list.filter'),
-			'filterRemove' => __('submission.list.filterRemove'),
+			'filter' => __('common.filter'),
+			'filterRemove' => __('common.filterRemove'),
```
Comment (bozana) [inline controllers/list/submissions/PKPSubmissionsListHandler.inc.php]: this locale key contain a variable, that is not provided here, or?

File: controllers/list/submissions/PKPSubmissionsListHandler.inc.php [hunk 4]
```diff
@@ -139,11 +121,12 @@ public function getConfig() {
 			'yes' => __('common.yes'),
 			'no' => __('common.no'),
 			'deleting' => __('common.deleting'),
+			'currentStage' => __('submission.list.currentStage'),
```
Comment (bozana) [inline controllers/list/submissions/PKPSubmissionsListHandler.inc.php]: this locale key contains a variable that is not provided here, or?

File: locale/eu_ES/editor.xml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE locale SYSTEM "../../dtd/locale.dtd">
+
+<!--
+  * locale/eu_ES/editor.xml
+  *
+  * Copyright (c) 2013-2017 Simon Fraser University
+  * Copyright (c) 2003-2017 John Willinsky
```
Comment (bozana) [inline locale/eu_ES/editor.xml]: year...

File: locale/hr_HR/editor.xml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE locale SYSTEM "../../dtd/locale.dtd">
+
+<!--
+  * locale/hr_HR/editor.xml
+  *
+  * Copyright (c) 2013-2017 Simon Fraser University
+  * Copyright (c) 2003-2017 John Willinsky
```
Comment (bozana) [inline locale/hr_HR/editor.xml]: year...

File: locale/id_ID/editor.xml [hunk 7]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE locale SYSTEM "../../dtd/locale.dtd">
+
+<!--
+  * locale/id_ID/editor.xml
+  *
+  * Copyright (c) 2013-2017 Simon Fraser University
+  * Copyright (c) 2003-2017 John Willinsky
```
Comment (bozana) [inline locale/id_ID/editor.xml]: year...

File: locale/nb_NO/editor.xml [hunk 8]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE locale SYSTEM "../../dtd/locale.dtd">
+
+<!--
+  * locale/nb_NO/editor.xml
+  *
+  * Copyright (c) 2014-2017 Simon Fraser University
+  * Copyright (c) 2003-2017 John Willinsky
```
Comment (bozana) [inline locale/nb_NO/editor.xml]: year...

File: locale/nl_NL/editor.xml [hunk 9]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE locale SYSTEM "../../dtd/locale.dtd">
+
+<!--
+  * locale/nl_NL/editor.xml
+  *
+  * Copyright (c) 2013-2017 Simon Fraser University
+  * Copyright (c) 2003-2017 John Willinsky
```
Comment (bozana) [inline locale/nl_NL/editor.xml]: year...

File: locale/tr_TR/editor.xml [hunk 10]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE locale SYSTEM "../../dtd/locale.dtd">
+
+<!--
+  * locale/tr_TR/editor.xml
+  *
+  * Copyright (c) 2013-2017 Simon Fraser University
+  * Copyright (c) 2003-2017 John Willinsky
```
Comment (bozana) [inline locale/tr_TR/editor.xml]: year...

File: locale/uk_UA/editor.xml [hunk 11]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE locale SYSTEM "../../dtd/locale.dtd">
+
+<!--
+  * locale/uk_UA/editor.xml
+  *
+  * Copyright (c) 2013-2017 Simon Fraser University
+  * Copyright (c) 2003-2017 John Willinsky
```
Comment (bozana) [inline locale/uk_UA/editor.xml]: year... and below...

File: plugins/importexport/native/filter/PKPAuthorNativeXmlFilter.inc.php [hunk 12]
```diff
@@ -99,7 +99,7 @@ function createPKPAuthorNode($doc, $author) {
 		$this->createOptionalNode($doc, $authorNode, 'url', $author->getUrl());
 		$this->createOptionalNode($doc, $authorNode, 'orcid', $author->getOrcid());
 
-		$this->createLocalizedNodes($doc, $authorNode, 'biography', $author->getBiography(null));
+		$this->createLocalizedNodes($doc, $authorNode, 'biography', $author->getBiography());
```
Comment (bozana) [inline plugins/importexport/native/filter/PKPAuthorNativeXmlFilter.inc.php]: here as well -- the biography was a localized element...

File: plugins/importexport/users/filter/PKPUserUserXmlFilter.inc.php [hunk 13]
```diff
@@ -95,8 +95,8 @@ function createPKPUserNode($doc, $user) {
 		$userNode->appendChild($doc->createElementNS($deployment->getNamespace(), 'email', htmlspecialchars($user->getEmail(), ENT_COMPAT, 'UTF-8')));
 		$this->createOptionalNode($doc, $userNode, 'url', $user->getUrl());
 		$this->createOptionalNode($doc, $userNode, 'orcid', $user->getOrcid());
-		if (is_array($user->getBiography(null))) {
-			$this->createLocalizedNodes($doc, $userNode, 'biography', $user->getBiography(null));
+		if (is_array($user->getBiography())) {
+			$this->createLocalizedNodes($doc, $userNode, 'biography', $user->getBiography());
```
Comment (bozana) [inline plugins/importexport/users/filter/PKPUserUserXmlFilter.inc.php]: and here...

File: classes/identity/Identity.inc.php [hunk 14]
```diff
@@ -254,11 +254,10 @@ function getLocalizedBiography() {
 
 	/**
 	 * Get author biography.
-	 * @param $locale string
 	 * @return string
 	 */
-	function getBiography($locale) {
-		return $this->getData('biography', $locale);
+	function getBiography() {
+		return $this->getData('biography');
```
[the remark this answers is not in the document]
Comment (NateWr) [reply classes/identity/Identity.inc.php]: Ah, you're right. I could have sworn that I looked at some other getters and thought they worked that way. But I'm looking now and seeing what you say...

File: classes/services/entityProperties/EntityPropertyInterface.inc.php [hunk 15]
```diff
@@ -26,8 +26,8 @@
 	public function getProperties($entity, $props, $args = null);
 
 	/**
-	 * Returns summary properties for a submission
-	 * @param Submission $submission
+	 * Returns summary properties for an entity
+	 * @param object $entity
```
[the remark this answers is not in the document]
Comment (NateWr) [reply classes/services/entityProperties/EntityPropertyInterface.inc.php]: > now I am wondering if we changed it ... ?

No, I'm just an idiot and do this in reverse all the time! :)

File: classes/services/queryBuilders/UserListQueryBuilder.inc.php [hunk 16]
```diff
@@ -222,17 +329,98 @@ public function get() {
...
+							->orWhere(function($q) use ($word) {
+								$q->where('us.setting_name', 'biography');
+								$q->where('us.setting_value', 'LIKE', "%{$word}%");
+							})
+							->orWhere(function($q) use ($word) {
+								$q->where('us.setting_name', 'orcid');
+								$q->where('us.setting_value', 'LIKE', "%{$word}%");
+							})
+							->orWhere('cves.setting_value', 'LIKE', "%{$word}%");
+					});
```
[the remark this answers is not in the document]
Comment (NateWr) [reply classes/services/queryBuilders/UserListQueryBuilder.inc.php]: 🙏

File: controllers/grid/users/reviewer/ReviewerGridRow.inc.php [hunk 17]
```diff
@@ -117,6 +117,27 @@ function initialize($request, $template = null) {
 					'more_info'
 				)
 			);
+
+			// Add gossip action when appropriate
+			$reviewAssignment = $this->getData();
```
[the remark this answers is not in the document]
Comment (NateWr) [reply controllers/grid/users/reviewer/ReviewerGridRow.inc.php]: 👍

File: controllers/list/users/SelectReviewerListHandler.inc.php (hunk 1 above)
Comment (NateWr) [reply controllers/list/users/SelectReviewerListHandler.inc.php]: I think so? I've put all of the new list handlers into `classes_controllers_list`, but I'm not sure if I'm doing that docblock right.

File: controllers/list/users/SelectReviewerListHandler.inc.php (hunk 2 above)
Comment (NateWr) [reply controllers/list/users/SelectReviewerListHandler.inc.php]: Good idea.

File: controllers/list/submissions/PKPSubmissionsListHandler.inc.php (hunk 3 above)
Comment (NateWr) [reply controllers/list/submissions/PKPSubmissionsListHandler.inc.php]: The Vue.js components include a `__()` mixin which can compile translations within Vue components. That way we can update strings based on the data loaded into the component:

https://github.com/pkp/ui-library/blob/master/src/mixins/global.js#L14-L45

**Comment by bozana** [comment]
closing this PR in favor of https://github.com/pkp/pkp-lib/pull/3411

**Pull request closed without merging**